src/algokit/cli/deploy.py

```diff
@@ -0,0 +1,108 @@
+import logging
+import os
+from pathlib import Path
+
+import click
+
+from algokit.core.constants import DEFAULT_NETWORKS, DEPLOYER_KEY, DISPENSER_KEY, LOCALNET, MAINNET
+from algokit.core.deploy import (
+    execute_deploy_command,
+    get_genesis_network_name,
+    load_deploy_command,
+    load_deploy_config,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def extract_mnemonics(skip_mnemonics_prompts: bool, network: str) -> tuple[str | None, str | None]:  # noqa: FBT001
+    """
+    Extract environment variables, prompt user if needed.
+
+    :param skip_mnemonics_prompts: A boolean indicating if user prompt should be skipped.
+    :param network: The name of the network ('localnet', 'testnet', 'mainnet', etc.)
+    :return: A tuple containing deployer_mnemonic and dispenser_mnemonic.
+    """
+    deployer_mnemonic = os.environ.get(DEPLOYER_KEY, "")
+    dispenser_mnemonic = os.environ.get(DISPENSER_KEY, None)
+
+    is_not_localnet = network != LOCALNET
+    is_deployer_mnemonic_empty = not skip_mnemonics_prompts and not deployer_mnemonic
+    is_dispenser_mnemonic_empty = not skip_mnemonics_prompts and not dispenser_mnemonic
+
+    if is_deployer_mnemonic_empty and is_not_localnet:
+        deployer_mnemonic = click.prompt("deployer-mnemonic", hide_input=True)
+
+    if (
+        is_dispenser_mnemonic_empty
+        and is_not_localnet
+        and click.confirm("Do you want to use a dispenser account?", default=False)
+    ):
+        dispenser_mnemonic = click.prompt("dispenser-mnemonic", hide_input=True)
```

Comment:
this whole structure using booleans is very confusing to me.

wouldn't it be simpler to structure with if statements?

like:
```
    if network != LOCALNET and not skip_mnemonics_prompts:
        if not deployer_mnemonic:
            deployer_mnemonic = click.prompt("deployer-mnemonic", hide_input=True)

        if not dispenser_mnemonic:
            use_dispenser = click.confirm("Do you want to use a dispenser account?", default=False)
            if use_dispenser:
                dispenser_mnemonic = click.prompt("dispenser-mnemonic", hide_input=True)
```

Reply:
Done 👍